src/main/java/ch/ivyteam/ivy/maven/engine/EngineClassLoaderFactory.java

```diff
@@ -97,7 +97,7 @@ public URLClassLoader createEngineClassLoader(File engineDirectory) throws IOExc
   {
     if (engineDirectory == null || !engineDirectory.isDirectory())
     {
-      return List.of();
+      throw new RuntimeException("The engineDirectory is missing");
```

Comment:
slightly better would be to include the reference to 'engineDirectory' in the exception message. Just in case the directory exists, but its of type file... we would get some help why it doesn't work.